python/setup.py

```diff
@@ -1,55 +1,91 @@
-            shutil.rmtree("src/bpcells-cpp")
-            open("src/bpcells-cpp", "wb").write(contents)
-
-extra_args = []
+import shutil
+for path in ["bpcells-cpp", "vendor"]:
+    if not os.path.islink(f"src/{path}") and not os.path.isdir(f"src/{path}"):
+        contents = open(f"src/{path}", "rb").read()
+        os.remove(f"src/{path}")
+        shutil.copytree(f"../r/src/{path}", f"src/{path}")
+        open(f"src/{path}/old-{path}", "wb").write(contents)
+
+class BuildHook(build):
+    def run(self): 
+        try:
+            build.run(self)
+        finally:
+            # Undo the manual copying of symlinks
+            for path in ["bpcells-cpp", "vendor"]:
+                if not os.path.islink(f"src/{path}") and os.path.isfile(f"src/{path}/old-{path}"):
+                    contents = open(f"src/{path}/old-{path}", "rb").read()
+                    shutil.rmtree(f"src/{path}")
+                    open(f"src/{path}", "wb").write(contents)
+
+extra_compile_args = []
 if "BPCELLS_DEBUG" in os.environ:
-    extra_args.append("-g")
+    extra_compile_args.append("-g")
 
 if "BPCELLS_NO_OPT" in os.environ:
-    extra_args.append("-O0")
+    extra_compile_args.append("-O0")
 else:
-    extra_args.append("-O2")
+    extra_compile_args.append("-O2")
 
 # Plan: take an environment variable for the libs
 # - HWY_INCLUDE_DIR: Include path for hwy library
 # - HWY_LIB_DIR: Lib path for hwy library
-include_dirs = ["src/bpcells-cpp"]
+libraries = ["hdf5", "hwy"]
+include_dirs = ["src/bpcells-cpp", "src/vendor"]
 library_dirs = []
-if "HWY_INCLUDE_DIR" in os.environ:
-    include_dirs.append(os.environ["HWY_INCLUDE_DIR"])
-if "HWY_LIB_DIR" in os.environ:
-    library_dirs.append(os.environ["HWY_LIB_DIR"])
+
+if platform.system() == "Windows":
+    extra_compile_args = ["/std:c++17"]
+    if "BPCELLS_DEBUG" in os.environ:
+        extra_compile_args.append("/Zi")
+
+    if "BPCELLS_NO_OPT" in os.environ:
+        extra_compile_args.append("/Od")
+    else:
+        extra_compile_args.append("/O2")
+
+    libraries.append("zlib")
+
+    # Mimic the unix compiler environment variable settings
+    if "CPATH" in os.environ:
+        include_dirs.extend(os.environ["CPATH"].split(";"))
+    if "LIBRARY_PATH" in os.environ:
+        library_dirs.extend(os.environ["LIBRARY_PATH"].split(";"))
+else:
+    extra_compile_args = [
+        "-std=c++17",
+        '-Wno-unused-but-set-variable',
+    ] 
+    if "BPCELLS_DEBUG" in os.environ:
+        extra_compile_args.append("-g")
+
+    if "BPCELLS_NO_OPT" in os.environ:
+        extra_compile_args.append("-O0")
+    else:
+        extra_compile_args.append("-O2")
+
+    libraries.append("z")
+
+    # Mimic the unix compiler environment variable settings
+    if "CPATH" in os.environ:
+        include_dirs.extend(os.environ["CPATH"].split(":"))
+    if "LIBRARY_PATH" in os.environ:
+        library_dirs.extend(os.environ["LIBRARY_PATH"].split(":"))
```

Comment:
Personal preference, but I think even if they are single use, it would be more clean to throw each task into separate functions.  ie one that does the improperly checked out symlinks, one that assigns `extra_compile_args`, `library_dirs`, `include_dirs` based on platform